timelock-impl/src/main/java/com/palantir/atlasdb/timelock/auth/api/User.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * (c) Copyright 2019 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.atlasdb.timelock.auth.api;
+
+import org.immutables.value.Value;
+
+@Value.Immutable
+public interface User {
+    User ANONYMOUS = User.of("anonymous");
+
+    String getUserName();
+    boolean isAdmin();
```

Comment:
Admin is a role, and we have only two roles, so we can do it with a boolean. Otherwise this would return an enum type describing the role.

Reply:
thinking again, I feel like it makes more sense to move admin knowledge to authorizer, to keep it in one place. Otherwise both authenticator and authorizer need to be aware of admin users.